Source/ProgressionSystemRuntime/Private/Components/PSSpotComponent.cpp

```diff
@@ -47,6 +47,50 @@ void UPSSpotComponent::OnReset_Implementation()
 	}
 }
 
+// Listen game states to switch character skin. 
+void UPSSpotComponent::OnGameStateChanged_Implementation(ECurrentGameState CurrentGameState)
+{
+	// it's possible that spot might not be loaded till that time so no ensure added
+	UPSSpotComponent* CurrentSpot = UPSWorldSubsystem::Get().GetCurrentSpot();
+
+	// check the spot  
+	if (CurrentSpot == nullptr || CurrentSpot != this)
+	{
+		return;
+	}
+
+	if (CurrentGameState == ECurrentGameState::GameStarting)
+	{
+		UMySkeletalMeshComponent& MeshComp = GetMeshChecked();
+		const int32 CurrentSkinIndex = MeshComp.GetAppliedSkinIndex();
+		bool isCurrentSkinAvailable = MeshComp.IsSkinAvailable(CurrentSkinIndex);
+
+		if (!isCurrentSkinAvailable)
+		{
+			for (int32 i = CurrentSkinIndex; i >= 0; i--)
```

Comment:
Please try to never name any variables with single character, it's not good practice in Unreal, please rename it to `Index` or `idx`. Or `count`, which is example from unreal coding standards:
```
for (int32 Count = 0; Count != 10; ++Count)
```

Reply:
done https://github.com/moinm3uw/ProgressionSystem/pull/63/commits/7606f298ecaa8553d7dd39297072d105e0a3421c